File: Interview-Prep/string/Character-filtering/242-optimized/242.js
var isAnagram = function(s, t) {
    if(s.length!=t.length) return false
    let frequency = Array(26).fill(0)
    for(let i=0; i<s.length; i++){
        frequency[s[i].charCodeAt()-97]++
        frequency[t[i].charCodeAt()-97]--
    }
    for(let i=0; i<26; i++){
        if(frequency[i]!=0) return false;
    }
    return true;
};
console.log(isAnagram("ann", "nan"))

File: Interview-Prep/string/Character-filtering/242-optimized/go/242-.go
package main

import "fmt"

func isAnagram(s string, t string) bool {
	if len(s)!= len(t){
		return false;
	}
    frequency := make([]int, 26)
	for i:=0; i<len(s); i++{
		frequency[s[i]-97]++;
		frequency[t[i]-97]--;
	}
	for i:=0; i<26; i++{
		if frequency[i]!=0{
			return false;
		}
	}
	return true;

}
func main(){
	fmt.Println(isAnagram("aina", "ania"));
}